### PR TITLE
feat: req/res logging for monitoring

### PR DESCRIPTION
**Pull request by FMFigueroa**

### CODE REVIEW

File: crates/file-explorer/src/fs.rs [hunk 1]
```diff
@@ -10,6 +10,7 @@ use axum::body::Bytes;
 use chrono::{DateTime, Local};
 use futures::Stream;
 use mime_guess::{from_path, Mime};
+#[allow(unused_imports)]
```
Comment (LeoBorai) [inline crates/file-explorer/src/fs.rs:13]: Please remove the unused import if it is complaining

```suggestion
```

File: crates/file-explorer/src/fs.rs [hunk 2]
```diff
@@ -28,6 +29,7 @@ pub struct File {
     pub metadata: Metadata,
 }
 
+#[allow(dead_code)]
```
Comment (LeoBorai) [inline crates/file-explorer/src/fs.rs:32]: Please remove

```suggestion
```

File: crates/file-explorer/src/fs.rs [hunk 3]
```diff
@@ -115,6 +117,7 @@ pub struct Directory {
 }
 
 /// An entry in a directory
+#[allow(dead_code)]
```
Comment (LeoBorai) [inline crates/file-explorer/src/fs.rs:120]:
```suggestion
```

File: crates/file-explorer/src/fs.rs [hunk 4]
```diff
@@ -133,7 +136,7 @@ pub async fn open(path: PathBuf) -> Result<Entry> {
 
     Ok(Entry::File(Box::new(File::new(path, file, metadata))))
 }
-
+#[allow(unused_variables)]
```
Comment (LeoBorai) [inline crates/file-explorer/src/fs.rs:139]:
```suggestion
```

File: crates/http-server/src/middleware/monitoring.rs [hunk 5]
```diff
@@ -0,0 +1,64 @@
+use axum::{body::Body, http::Request, response::Response};
+use futures_util::future::BoxFuture;
+use std::task::{Context, Poll};
+use std::time::Instant;
+use tower::{Layer, Service};
```
Comment (LeoBorai) [inline crates/http-server/src/middleware/monitoring.rs:5]: Rearrange following convention: `std`, external, crate, super, self

```suggestion
use std::task::{Context, Poll};
use std::time::Instant;

use axum::{body::Body, http::Request, response::Response};
use futures_util::future::BoxFuture;
use tower::{Layer, Service};
```

File: crates/http-server/src/middleware/monitoring.rs [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    fn call(&mut self, request: Request<Body>) -> Self::Future {
+        let start_time = Instant::now();
+        let request_info = RequestInfo {
+            method: request.method().to_string(),
+            path: request.uri().path().to_string(),
+        };
+
+        let future = self.inner.call(request);
```
Comment (LeoBorai) [inline crates/http-server/src/middleware/monitoring.rs:47]: Nitpick!

```suggestion
        let start_time = Instant::now();
        let request_info = RequestInfo {
            method: request.method().to_string(),
            path: request.uri().path().to_string(),
        };
        let future = self.inner.call(request);

```

File: crates/http-server/src/middleware/monitoring.rs [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        self.inner.poll_ready(cx)
+    }
+
+    fn call(&mut self, request: Request<Body>) -> Self::Future {
+        let start_time = Instant::now();
+        let request_info = RequestInfo {
+            method: request.method().to_string(),
+            path: request.uri().path().to_string(),
+        };
```
Comment (LeoBorai) [inline crates/http-server/src/middleware/monitoring.rs:45]: I think we can use `request.method().to_string()` and `request.uri().path().to_string()` straight!
No need to allocate `request_info` to drop it just after L53

https://github.com/http-server-rs/http-server/pull/326/files#diff-11aa8b5243fe114bb1329d03faa4979762046503a2fd2493683f1bdeeed12adbR53-R59

File: crates/file-explorer/src/fs.rs (hunk 1 above)
Comment (FMFigueroa) [reply crates/file-explorer/src/fs.rs:13]: I don't know why there is dead code in the crate file-explorer but the warnings are present.

File: crates/file-explorer/src/fs.rs (hunk 2 above)
Comment (FMFigueroa) [reply crates/file-explorer/src/fs.rs:32]: I don't know why there is dead code in the crate file-explorer but the warnings are present.

File: crates/file-explorer/src/fs.rs (hunk 3 above)
Comment (FMFigueroa) [reply crates/file-explorer/src/fs.rs:120]: I don't know why there is dead code in the crate file-explorer but the warnings are present.

File: crates/file-explorer/src/fs.rs (hunk 4 above)
Comment (FMFigueroa) [reply crates/file-explorer/src/fs.rs:139]: I don't know why there is dead code in the crate file-explorer but the warnings are present.

File: crates/http-server/src/middleware/monitoring.rs (hunk 7 above)
Comment (FMFigueroa) [reply crates/http-server/src/middleware/monitoring.rs:45]: This option is not possible straight!

File: crates/http-server/src/middleware/monitoring.rs (hunk 5 above)
Comment (FMFigueroa) [reply crates/http-server/src/middleware/monitoring.rs:5]: ok!

File: crates/http-server/src/middleware/monitoring.rs (hunk 7 above)
Comment (LeoBorai) [reply crates/http-server/src/middleware/monitoring.rs:45]: I wonder why? 🤔

In the docs I can see that `uri` and `method` borrow `self`:

https://docs.rs/http/0.2.9/http/request/struct.Request.html#method.method

https://docs.rs/http/0.2.9/http/request/struct.Request.html#method.uri

So it shouldn't be moving. Can you share the compile error please?

Comment (LeoBorai) [reply crates/http-server/src/middleware/monitoring.rs:45]: Another improvement I see is saving the call to `to_string`, given that both:

- Uri: https://docs.rs/http/0.2.9/http/uri/struct.Uri.html#impl-Display-for-Uri
- Method: https://docs.rs/http/0.2.9/http/method/struct.Method.html#impl-Display-for-Method

Implement display.

File: crates/file-explorer/src/fs.rs (hunk 1 above)
Comment (LeoBorai) [reply crates/file-explorer/src/fs.rs:13]: I can check locally and push a fix! Thanks!

File: crates/http-server/src/middleware/monitoring.rs (hunk 7 above)
Comment (FMFigueroa) [reply crates/http-server/src/middleware/monitoring.rs:45]: I think wrapping the request in an Arc can be an option like this:
https://github.com/http-server-rs/http-server/blob/main/src/server/middleware/logger.rs
But doing it, I think I'm going outside the scope of the issue.

**Comment by LeoBorai** [comment]
@FMFigueroa I have fixed issues with outdated branch on [`http-server-rs:integration/axum`](https://github.com/http-server-rs/http-server/tree/integration/axum) against HEAD. Can you pull in your local please?

**Pull request closed without merging**